lib/tile.js

```diff
@@ -158,29 +142,33 @@ Tile.prototype.containsPoint = function(x,y) {
 }
 
 Tile.prototype.sameTypeAs = function(tile) {
-	return this.type == tile.type;
+	if(tile) { 
+		return this.type == tile.type;
+	} else {
+		return false;
+	}
 }
 
 Tile.prototype.possibleMatch = function() {
-	if(this.tileBelow() && this.tileBelow().sameTypeAs(this)) {
```

Comment:
You could argue that this refactor actually makes the code a little bit harder to understand, even though you've decreased the total lines of code. Now the coder needs to know what relativeTile means to scan the code. I think you're on the right track with trying to break this section down though!